docs/details.md

```diff
@@ -0,0 +1,173 @@
+
+**Table of Contents**
+
+<!--ts-->
+   * [elixir-omg applications](#elixir-omg-applications)
+      * [Child chain server](#child-chain-server)
+         * [Using the child chain server's API](#using-the-child-chain-servers-api)
+            * [HTTP-RPC](#http-rpc)
+         * [Running a child chain in practice](#running-a-child-chain-in-practice)
+            * [Ethereum private key management](#ethereum-private-key-management)
+               * [geth](#geth)
+               * [parity](#parity)
+            * [Specifying the fees required](#specifying-the-fees-required)
+            * [Managing the operator address](#managing-the-operator-address)
+               * [Nonces restriction](#nonces-restriction)
+               * [Funding the operator address](#funding-the-operator-address)
+      * [Watcher](#watcher)
+         * [Modes of the watcher](#modes-of-the-watcher)
+         * [Using the watcher](#using-the-watcher)
+         * [Endpoints](#endpoints)
+         * [Ethereum private key management](#ethereum-private-key-management-1)
+
+
+# `elixir-omg` applications
+
+`elixir-omg` is an umbrella app comprising of several Elixir applications:
+
+The general idea of the apps responsibilities is:
+  - `omg` - common application logic used by both the child chain server and watcher
+  - `omg_bus` - an internal event bus to tie services together
+  - `omg_child_chain` - child chain server
+    - tracks Ethereum for things happening in the root chain contract (deposits/exits)
+    - gathers transactions, decides on validity, forms blocks, persists
+    - submits blocks to the root chain contract
+    - see `apps/omg_child_chain/lib/omg_child_chain/application.ex` for a rundown of children processes involved
+  - `omg_child_chain_rpc` - an HTTP-RPC server being the gateway to `omg_child_chain`
+  - `omg_db` - wrapper around the child chain server's database to store the UTXO set and blocks necessary for state persistence
+  - `omg_eth` - wrapper around the [Ethereum RPC client](https://github.com/exthereum/ethereumex)
+  - `omg_performance` - performance tester for the child chain server
+  - `omg_status` - application monitoring facilities
+  - `omg_utils` - various non-omg-specific shared code
+  - `omg_watcher` - the [Watcher](#watcher)
+  - `omg_watcher_info` - the [Watcher Info](#watcher)
+  - `omg_watcher_rpc` - an HTTP-RPC server being the gateway to `omg_watcher`
+
+See [application architecture](docs/architecture.md) for more details.
+
+## Child chain server
+
+`:omg_child_chain` is the Elixir app which runs the child chain server, whose API is exposed by `:omg_child_chain_rpc`.
+
+For the responsibilities and design of the child chain server see [Tesuji Plasma Blockchain Design document](docs/tesuji_blockchain_design.md).
```

Comment:
maybe it's time to rename this to `blockchain_design.md` and scrape the `tesuji` part, since we're doing this (very needed :pray:) makeover. Maybe we could keep the `docs/tesuji...md` with just the link to the new location, so that references if they exist don't lead to something.